tests/test_itemized.py

```diff
@@ -106,15 +107,29 @@ def test_filter(self):
         self.assertEqual(len(results), 1)
         self.assertEqual(results[0]['contributor_state'], 'CA')
 
-    def test_filter_zip(self):
+    def test_filter_sched_a_zip(self):
         [
             factories.ScheduleAFactory(contributor_zip=96789),
-            factories.ScheduleAFactory(contributor_zip=66111)
+            factories.ScheduleAFactory(contributor_zip=9678912),
+            factories.ScheduleAFactory(contributor_zip=967891234)
         ]
 
         results = self._results(api.url_for(ScheduleAView, contributor_zip=96789, **self.kwargs))
-        self.assertEqual(len(results), 1)
-        self.assertEqual(results[0]['contributor_zip'], '96789')
+        self.assertEqual(len(results), 3)
+        
+        results = self._results(api.url_for(ScheduleAView, contributor_zip=9678912, **self.kwargs))
```

Comment:
Even though this is what we discussed, it just feels wrong to me as far as the API is concerned. We should do one of:
1. Accept only (exactly) 5-digit zip codes for searches and return rows that have those 5 digits as a prefix.
2. Accept 5-digit zip codes for searches and return rows that have those 5 digits as a prefix. If 9 digits are given, do an exact match on those 9 digits. 

As the extra 4 digits in ZIP+4 is a mailing route that can change, don't accept 6, 7 or 8 digits.

Pinging @ccostino, @lbeaufort, @fecjjeng, @rjayasekera, @fec-jli .

Reply:
@vrajmohan - Option 1 would be my preference. Do you envision throwing validation if an API user supplies a zip over 5 digits? We do have some weirdness with 400 validation errors showing up as 500 errors, so that might be misleading.

For option 2, here's why I wouldn't recommend it:
**Data is inconsistent**
1 | Laura | 999 E St | 20463
2 | Laura | 999 E St | 204630923

**User searching for 204630923 would expect it to return:**
1 | Laura | 999 E St | 204630923
2 | Laura | 999 E St | 204630923

**..But because of inconsistent data, it only returns:**
2  | Laura  | 999 E St | 204630923

**and is missing**
1 | Laura | 999 E St | 20463
**...even though that address is in that area**

Reply:
I also like Option 1. Yes, I would throw a validation error for any zip other than 5 digits.

Reply:
Sounds good to me. Here's the issue for the 400 errors - looks like it's tagged for this 5.3. https://github.com/fecgov/openFEC/issues/2962

Reply:
This is more like product feature to me. Maybe we should get stakeholders to weigh in on this issue. Would there be case when user enters other than 5-digit zip?  
If user enters 123, the API will return sets starting with 123
if user enter 123456789, the API will return sets starting with 12345

maybe validation for minimum digit entered for zip instead?

@PaulClark2 @jwchumley